File: judge/java/tests/three_increasing_elements.java
package tests;
import java.util.*;
import java.lang.*;
import java.io.*;
import judge.*;
import datastruct.*;

public class three_increasing_elements {
    public static int num_test = 120;
    public static int[][] in_0;
    public static int[][] in_org_0;
    public static int[][] out;
public static boolean test(int[] answer, int i) {
    if(out[i].length != answer.length) return false;
    if(out[i][0] == -1 && out[i][1] == -1 && out[i][2] == -1) {
        return answer[0] == -1 && answer[1] == -1 && answer[2] == -1;
    } else {
        if(answer[0] == -1 || answer[1] == -1 || answer[2] == -1) {
            return false;
        } else {
            return (in_org_0[i][answer[0]] < in_org_0[i][answer[1]] && in_org_0[i][answer[1]] < in_org_0[i][answer[2]]) && (answer[0] < answer[1] && answer[1] < answer[2]);
        }
    }
}

    public static void load_test() {
        File fil = new File("judge/tests/three-increasing-elements.txt");
        FileReader inputFil = null;
        try {
            inputFil = new FileReader(fil);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        BufferedReader in = new BufferedReader(inputFil);
        try {
            in_0 = common.read_int_matrix(in);
            in_org_0 = test_common.copy(in_0);
            out = common.read_int_matrix(in);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void judge() {
        load_test();
        Solution s = new Solution();

        long startTime = System.currentTimeMillis();

        for(int i = 0; i < num_test; ++i) {
            int[] answer = s.three_increasing_nums(in_0[i]);
            if(!three_increasing_elements.test(answer,i)) {
                System.out.printf("%d / %d;", i+1, num_test);
                String outs = common.to_string(three_increasing_elements.in_org_0[i]);
                System.out.print(outs + ";");
                System.out.print(common.to_string(answer) + ";");
                System.out.println(common.to_string(out[i]));
                return;
            }
        }

        long estimatedTime = System.currentTimeMillis() - startTime;
        System.out.print("Accepted;");
        System.out.println(estimatedTime);
    }
}

File: judge/lua/tests/three-increasing-elements.lua
require("../solution")

local num_test = 120
local in_0 = {}
local in_org_0 = {}
local out = {}
function test(answer, i)
    if #out[i] ~= #answer then return false end

    if out[i][1] == -1 and out[i][2] == -1 and out[i][3] == -1 then
        return answer[1] == -1 and answer[2] == -1 and answer[3] == -1
    else
				out[i][1], out[i][2], out[i][3] = out[i][1]+1, out[i][2]+1, out[i][3]+1
        if answer[1] == -1 or answer[2] == -1 or answer[3] == -1 then
            return false
        else
            return (in_org_0[i][answer[1]] < in_org_0[i][answer[2]] and in_org_0[i][answer[2]] < in_org_0[i][answer[3]]) and (answer[1] < answer[2] and answer[2] < answer[3])
		end
	end
end

function load_test()
    local f = io.open("./judge/tests/three-increasing-elements.txt", "r")
    in_0 = read_num_matrix(f)
    in_org_0 = copy(in_0)
    out = read_num_matrix(f)
    f:close()
end

function judge()
    load_test()

    local start = os.clock()
    for i = 1, num_test do
        local answer = three_increasing_nums(in_0[i]) 
        if not test(answer, i) then
            io.write(string.format("%d / %d;", i, num_test))
            io.write(to_string(in_org_0[i]))
            io.write(";")
            io.write(to_string(answer))
            io.write(";")
            io.write(to_string(out[i]))
            io.write("\n")
            return
        end
    end

    local elapsed = math.floor((os.clock() - start) * 1000)
	print("Accepted;" .. elapsed)
end
